js/src/behaviors/tabs.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * Tab control for assessment lens
+ * @param container - Container of tab control component
+ */
+/**
+ * Tabs behaviors
+ */
+export function initTabs() {
+	initTabNavClickListeners();
+	initTabControlClickListeners();
+
+	// update when tabs into the view
+	const tabContainers = Array.from(document.querySelectorAll('tab-container')) as HTMLElement[];
+	for (const tc of tabContainers) {
+		updateTabNav(tc);
+	}
+}
+
+function initTabNavClickListeners() {
+	window.addEventListener('click', (event: Event) => {
+		const target =
+			event.target instanceof Element &&
+			(event.target.closest('[data-tab-nav]') as HTMLButtonElement);
+
+		if (!target) {
+			return;
+		}
+
+		event.preventDefault();
+		const container = event.target.closest('tab-container') as HTMLElement;
+		const tabs = Array.from(container.querySelectorAll('.tab-control')) as HTMLButtonElement[];
+
+		// get current activated tab
+		const currentActiveButton = container.querySelector(
+			'[aria-selected="true"]'
+		) as HTMLButtonElement;
+		const index = parseInt(currentActiveButton?.dataset.tabControl as string);
+
+		// update tab
+		if (index > 1 && target.dataset.tabNav === 'previous') {
+			updateTabState(container, currentActiveButton, index - 1);
+			updateTabPanel(container, index, index - 1);
+		} else if (index < tabs.length && target.dataset.tabNav === 'next') {
+			updateTabState(container, currentActiveButton, index + 1);
+			updateTabPanel(container, index, index + 1);
+		}
+		updateTabNav(container);
+	});
+}
+
+function initTabControlClickListeners() {
+	window.addEventListener('click', (event: Event) => {
+		const target =
+			event.target instanceof Element &&
+			(event.target.closest('.tab-control') as HTMLButtonElement);
+
+		if (!target) {
+			return;
+		}
+
+		event.preventDefault();
+		const container = event.target.closest('tab-container') as HTMLElement;
+		updateTabNav(container);
+	});
+	window.addEventListener('keydown', (event: KeyboardEvent) => {
+		if (event.key !== 'ArrowLeft' && event.key !== 'ArrowRight') {
```

Comment:
I think down and up also need to work here.

Reply:
when I was test this part i found the up and down didnt do anything

Reply:
Per [ARIA Authoring Practices guidance](https://www.w3.org/WAI/ARIA/apg/patterns/tabpanel/), up and down arrows aren't required to do anything (and arguably shouldn't do anything) when the tabs are arranged horizontally:

> 3. If the tab list is horizontal, it does not listen for `Down Arrow` or `Up Arrow` so those keys can provide their normal browser scrolling functions even when focus is inside the tab list.

Reply:
This is one of those cases where, again, I think I'd be happy either way.

I think we should do one _or_ the other here:
1. Allow Up and Down arrows to change tabs
2. Ensure the `div[role=tablist]` also has `aria-orientation="horizontal"` to indicate to screenreader users that these tabs are arranged horizontally, and that they can use left/right arrows to cycle through them.

Reply:
added `aria-orientation="horizontal" `

Reply:
@jim-benson had a question about this. If the `tab-container-element ` includes the required keyboard navigation, then why are we duplicating it here? It looks to me like it isn't being used.

Reply:
(We'll need some separate JS for things like the nav buttons, but not for the inner stuff related to tab-container-element)

Reply:
we are  doing it here cause we have arrow buttons as well and when i press an arrow key i need to check if the nav button needs to be hidden